documents/bootstrapper.go

```diff
@@ -1,8 +1,14 @@
 package documents
 
+// BootstrappedRegistry is the key to ServiceRegistry in Bootstrap context
+const BootstrappedRegistry = "BootstrappedRegistry"
+
 type Bootstrapper struct{}
 
 // Bootstrap sets the required storage and registers
-func (*Bootstrapper) Bootstrap(context map[string]interface{}) error {
+func (Bootstrapper) Bootstrap(ctx map[string]interface{}) error {
+	ctx[BootstrappedRegistry] = &ServiceRegistry{
```

Comment:
Why not calling NewServiceRegistry() here?

Reply:
good catch!